### ISSUE TITLE
[BUG] `import xxx.yyy as zzz` broken with language_level=2 (regression from 0.29.x)

### ISSUE THREAD

**Issue by da-woods** [cython/cython#5308]
### Describe the bug

See code...

### Code to reproduce the behaviour:

```cython
# testmod/cabc.pyx

import collections.abc as collections_abc
print(collections_abc)
```

Add an empty `__init__.py` to testmod (not sure if this is absolutely needed)

Compile with `python3 cythonize.py -2if testmod/cabc.pyx && python3 -c "import testmod.cabc"`

On the current master this print the `collections` module

### Expected behaviour

It should print the `collections.abc` module (and it does on the 0.29 branch)

### OS

Linux

### Python version

>3

### Cython version

current master

### Additional context

Bisected to 4993ba6a0194c2fbcf438ae2833f43c33781148d

**Comment by scoder** [comment]
This seems a critical bug that we should fix for beta-2. I expect quite a number of projects to keep their existing code base on `language_level=2` at least in the beginning. It should be a viable migration path.

**Comment by da-woods** [comment]
Agree. I'm actually quite surprised that this has been missed until now, suggesting few people are using `language_level=2` explicitly. But quite a few "bugs" on the fedora list looked like they'd be solved by language_level 2

**Comment by scoder** [comment]
Well, there is a test for this that runs in both language levels:
https://github.com/cython/cython/blob/master/tests/run/importas.pyx
What seems to be missing is the same thing from within a package. It's a bit tricky now that there are two implementations (`__Pyx_Import()` and `__Pyx_ImportDottedModule()`) which both seem needed for the LL2 case since it combines that Py2 preference for relative imports with an absolute import fallback.

I'm looking into it.
